cmd/bucket-handlers.go

```diff
@@ -481,6 +481,24 @@ func (api objectAPIHandlers) DeleteMultipleObjectsHandler(w http.ResponseWriter,
 		}
 
 		if replicateDeletes || hasLockEnabled || hasLifecycleConfig {
+			opts, err := delOpts(ctx, r, bucket, object.ObjectName)
+			if err != nil {
+				apiErr := toAPIError(ctx, err)
+				dErrs[index] = DeleteError{
+					Code:      apiErr.Code,
+					Message:   apiErr.Description,
+					Key:       object.ObjectName,
+					VersionID: object.VersionID,
+				}
+				continue
+			}
```

Comment:
this code block is unnecessary. To do a GetObjectInfo you will just need to pass VersionID in the ObjectOptions
```
opts := ObjectOptions{VersionID: object.VersionID}
if opts.VersionSuspended && opts.VersionID == "" {
    opts.VersionID = nullVersionID
}
goi, gerr = getObjectInfoFn(ctx, bucket, object.ObjectName, opts)
```